data_usage.go

```diff
@@ -0,0 +1,148 @@
+	}
+	defer fileCreator.Close()
+
+	start := time.Now()
+
+	conn, err := ConnectToSnowflake(config.Parameters, "")
+	if err != nil {
+		return data_usage.DataUsageSyncResult{Error: api.ToErrorResult(err)}
+	}
+	defer conn.Close()
+
+	// TODO: should be configurable
+	numberOfDays := 7
+	startDate := time.Now().AddDate(0, 0, -numberOfDays)
+	dateFormat := "2006-01-02"
+
+	historyQuery := fmt.Sprintf("SELECT QUERY_ID, EXECUTION_STATUS, QUERY_TEXT, DATABASE_NAME, SCHEMA_NAME, USER_NAME, START_TIME, END_TIME, EXECUTION_TIME, OUTBOUND_DATA_TRANSFER_BYTES, EXTERNAL_FUNCTION_TOTAL_SENT_ROWS FROM SNOWFLAKE.ACCOUNT_USAGE.QUERY_HISTORY WHERE start_time >= '%s'", startDate.Format(dateFormat))
+	dataUsageQuery := historyQuery
+
+	logger.Info(fmt.Sprintf("Retrieving query log from Snowflake with query: %s", dataUsageQuery))
+	rows, err := QuerySnowflake(conn, dataUsageQuery)
+	if err != nil {
+		return data_usage.DataUsageSyncResult{Error: api.ToErrorResult(err)}
+	}
+
+	var returnedRows []queryDbEntities
+
+	logger.Info("Scanning query results")
+	err = scan.Rows(&returnedRows, rows)
+	if err != nil {
+		logger.Error("Error scanning results into objects")
+		return data_usage.DataUsageSyncResult{Error: api.ToErrorResult(err)}
+	}
+
+	sec := time.Since(start).Round(time.Millisecond)
+	logger.Info(fmt.Sprintf("Fetched %d rows from Snowflake in %s", len(returnedRows), sec))
+
+	logger.Info("Preparing import file")
+	timeFormat := "2006-01-02T15:04:05.999999-07:00"
+	executedStatements := make([]dub.Statement, 0, 20)
+	for _, returnedRow := range returnedRows {
+		startTime, err := time.Parse(timeFormat, returnedRow.StartTime)
+		if err != nil {
+			logger.Error(fmt.Sprintf("Error parsing start time of '%s', expected format is: '%s'", returnedRow.StartTime, timeFormat))
+		}
+		endTime, err := time.Parse(timeFormat, returnedRow.EndTime)
+		if err != nil {
+			logger.Error(fmt.Sprintf("Error parsing end time of '%s', expected format is: '%s'", returnedRow.StartTime, timeFormat))
+		}
+
+		databaseName := ""
+		if returnedRow.DatabaseName.Valid {
+			databaseName = returnedRow.DatabaseName.String
+		}
+		schemaName := ""
+		if returnedRow.SchemaName.Valid {
+			schemaName = returnedRow.SchemaName.String
+		}
+		accessDataObjects := common.ExtractInfoFromQuery(returnedRow.Query, databaseName, schemaName)
+
+		if len(accessDataObjects) == 0 {
+			logger.Info(fmt.Sprintf("No data objects returned for query: %s", returnedRow.Query))
+		}
+
+		du := dub.Statement{
+			ExternalId:          returnedRow.ExternalId,
+			AccessedDataObjects: accessDataObjects,
+			Status:              returnedRow.Status == "SUCCESS",
+			User:                returnedRow.User,
+			StartTime:           startTime.UnixMilli(),
+			EndTime:             endTime.UnixMilli(),
+			TotalTime:           returnedRow.TotalTime,
+			BytesTransferred:    returnedRow.BytesTranferred,
+			RowsReturned:        returnedRow.RowsReturned,
+		}
+		executedStatements = append(executedStatements, du)
```

Comment:
Ok, the ORDER BY (for pagination is necessary): https://docs.snowflake.com/en/sql-reference/constructs/limit.html#:~:text=An%20ORDER%20BY%20clause%20is,use%20an%20ORDER%20BY%20clause. 
```
An [ORDER BY](https://docs.snowflake.com/en/sql-reference/constructs/order-by.html) clause is not required; however, without an [ORDER BY](https://docs.snowflake.com/en/sql-reference/constructs/order-by.html) clause, the results are non-deterministic because results within a result set are not necessarily in any particular order. To control the results returned, use an [ORDER BY](https://docs.snowflake.com/en/sql-reference/constructs/order-by.html) clause.
```